### ISSUE TITLE
Mirroring Posts options

### ISSUE THREAD

**Issue by tiotrom** [friendica/friendica#9592]
The mirror posts as your own feature is amazing. Thank you! Some suggestions:

1. Allow it for all kinds of contacts. Not only RSS and Twitter.
2. Allow for keyword filtering. "Exclude All" and "Include All". So one has control over it. Say I can choose to only show posts that contain the words "automation, 3d printing, foss" but I do not want posts containing "COVID, politics, trump". That'd be fantastic to have.

**Comment by annando** [comment]
I don't really like to allow it for all types. Then there is the danger that people just use Friendica like they would use  for example IFTTT. Means: Most likely there would be a lot of people who would use this to mirror their Mastodon posts to Twitter. And this would create ghost accounts where no one answers when you reply to them on their Friendica account.

Mirroring is great for networks that aren't spoken natively by Friendica. But I wouldn't want to implement it for networks that Friendica federates.

This thing with the allow- and denylist however sounds great.

**Comment by MrPetovan** [comment]
Same feeling here, and I would be more precise and say that mirroring is great for networks _that aren't supported natively by other federated software platforms_. Meaning that you can't tell Mastodon or Diaspora contacts to subscribe to one of your RSS feeds or your Twitter account directly from their accounts, while you can for your additional Friendica accounts for example.

**Comment by tiotrom** [comment]
I get you points guys.  I think the allow/deny keywords thing could be very useful tho. 

> Most likely there would be a lot of people who would use this to mirror their Mastodon posts to Twitter.

They can easily do it now tho. Add your Mastodon RSS, then mirror as your own posts, then select to auto-post on Twitter, right? And they can do that for PeerTube, and other Federated networks that support an RSS. 

What I had in mind was this: we are a few people working on our TROM project and some of us do their independent work. We have 1 Friendica main page, and I wanted to mirror the posts made by some of our team members who use PeerTube, Mastodon or other platforms, to our Friendica page, as if our page posts these entries. I can still do that via RSS, but is a bit cumbersome and at times (PeerTube) may not look like a "natural" post since it uses the RSS. You know what I mean?

**Comment by MrPetovan** [comment]
> They can easily do it now tho. Add your Mastodon RSS, then mirror as your own posts, then select to auto-post on Twitter, right? And they can do that for PeerTube, and other Federated networks that support an RSS.

Yes but there is an extra step required, the Mastodon RSS feed isn't immediately available, so it is a hurdle that prevents most or at least some of the unwanted behavior.

**Comment by tiotrom** [comment]
hm...I doubt this has any impact, or significant at all. Look at me: I discovered Friendica a month or so ago. I had a Mastodon account for 2 years. It took me a few hours to understand how to mirror from one to another. If people want to do this, they will find a way very quickly. I am wondering why would people want to use Friendica as a connector from Twitter-to-Mastodon, when there are lots of such alternatives out there that do just that? 

The issue here is with Twitter as far as I understand since Friendica users can auto-post on Twitter... else if people can mirror Mastodon or PeerTube accounts to Friendica, then they do that because they use Friendica and want the activity to be there, right?

In other words, you are worried that some users may choose to mirror certain Mastodon accounts, or PeerTube, or Pleroma, and so forth, to their Friendica, to then have these posts auto-posted on Twitter? If so, what if you add this an an option on Friendica, and say if "mirroring" is enabled, then auto-post on Twitter is disabled? You can't have both at the same time.

**Comment by tiotrom** [comment]
So :) I suggest this option:

If mirroring is enabled, then auto-post on Twitter is disabled.
If auto-post on Twitter is enabled, the mirroring is disabled.

And with this, you avoid the above issue. Right?

**Comment by MrPetovan** [comment]
From experience, I know that making something more accessible automatically increases usage. Because not everyone has the idea to do it in the first place, and among people who have the idea, not everyone is ready to get to the very bottom of it. So we won't go out of our way to prevent mirroring from specifically a Mastodon RSS feed even though we probably could, but we won't allow mirroring from regular Mastodon contacts either.

Beyond that, it's a philosophy question, what behavior do we want to encourage or deter, independently of the actual feasibility.

**Comment by tiotrom** [comment]
I understand you, but maybe the suggestion I made above, can deal with such questions. You give people more options, and stop them from abusing the platform, since they can't autopost on twitter if mirroring is enabled. What made me move from Mastodon to Friendica, was the ability to mirror RSS, and follow RSS or other federated networks. That's amazing. Mirroring from other accounts could also be a plus.

**Comment by annando** [comment]
I do completely understand you. I have to think of a good compromise. One would be to allow the mirroring - but to stok with it when the user hadn't logged in for a week.

**Comment by annando** [comment]
I just had got another idea: What about - instead of to mirror - to add an option to automatically reshare an account?

**Comment by AlfredSK** [comment]
Umm... Isn't that the same thing? Just curious about the difference.

Edit: ...thinking... Is it create native reshares vs. create new posts (with same content)?

**Comment by annando** [comment]
It is like pressing the new "share" button.

**Comment by AlfredSK** [comment]
Thank you. That's what I meant with "create native reshares". And it is a good idea.

**Comment by tiotrom** [comment]
Aham, so to auto create reshares from accounts you mean? Instead of mirroring to reshare automatically?

**Comment by annando** [comment]
Yeah, The resharing has got the advantage that comments to this post will automatically reach the original poster. And people will get to know the original account as well.

**Comment by tiotrom** [comment]
Sounds very good actually! I really like it! Great idea.

**Comment by AlfredSK** [comment]
I just made two tests...

1. Post from a Misskey user. Auto-reshare enabled for that contact in my personal Friendica account. Worked. 🎉
2. Post from a Mastodon user. Auto-reshare enabled for that contact in a Friendica forum account. Didn't work. 🤔

Is there something special with forum accounts ~~(e.g. uni-directional vs. bi-directional relationship)~~ that prevents this from working?

What I'm actually trying to accomplish is when I post an announcement on my Mastodon account the local support forum on Friendica should auto-reshare that post with its followers.

Edit: Can't be the relationship type. The forum has a bi-directional relationship with the Mastodon user.

**Comment by annando** [comment]
Forum accounts are special. So I can imagine that it can't work.

**Comment by SpcCw** [comment]
I would like to second the request for keywords denylist for RSS feeds (meaning "if the keyword is found in the RSS item, it won't be published"). I am not sure if allowlist is needed, can't think of any usecase for that.

Friendica is great for broadcasting RSS feeds to Fediverse and it can be a lot of fun to discuss e.g. local news with friends right on social network, refer to some item or reshare it - however since RSS comes from external source a little control over it is very desirable.

**Comment by tio-trom** [comment]
> I am not sure if allowlist is needed, can't think of any usecase for that.

Well we have a Friendica page for our project but we are a few project members. We often post updates and would be great if only the updates will go to our common Friendica page. For example we can use the hashtag #tromupdates when we post an update on our profiles. Then I can setup our Friendica page to ONLY repost posts from these accounts if the word "tromupdates" is found. This way we can smartly push updates to our Friendica page.

This would be fantastic not only for such cases, but also to filter OUT certain things from an RSS or other contacts that you may want to mirror. Please Friendica friends, implement this :D

**Comment by SpcCw** [comment]
Actually now I agree with @tiotrom that allowlist also wouldn't hurt :) 

I host quite a bit of mirrored RSS accounts from various sources some of which are getting popular across Fediverse. While denylist is absolutely essential because sometimes weird stuff creeps in I get requests for some channels where it would be nice to filter items for specific tags.

Having such functionality would make Friendica even more of Swiss pocket knife for Fedi than it already is :)

**Comment by sjn** [comment]
This feature would be great to have! :+1:
